pkg/compact/compact_test.go

```diff
@@ -174,3 +178,358 @@ func BenchmarkGatherNoCompactionMarkFilter_Filter(b *testing.B) {
 	}
 
 }
+
+type planResult struct {
+	compactionBlocks, compactionRuns float64
+}
+
+func TestPlanSimulate(t *testing.T) {
+	reg := prometheus.NewRegistry()
+	unRegisterer := &receive.UnRegisterer{Registerer: reg}
+
+	extLabels := labels.FromMap(map[string]string{"a": "1"})
+
+	for _, tcase := range []struct {
+		testName    string
+		testPlanner *tsdbBasedPlanner
```

Comment:
I see we are just using the same TSDB planner and creating the planner every time. If you just plan to use the same tsdb planner you can just remove it from the testcase and create a single TSDB planner at the beginning.

Reply:
I think it would be good if we can also create a mock planner for testing

Reply:
Thanks! Re-used a single planner in the test cases now.

Reply:
Can you also add a mock planner then?

Reply:
What exactly is a mock planner? I'm not fully clear.
Thanks!